packages/main/src/MenuItem.ts

```diff
@@ -104,14 +107,25 @@ type MenuItemAccessibilityAttributes = Pick<AccessibilityAttributes, "ariaKeySho
  * @since 1.10.0
  */
 @event("close")
+
+/**
+ * Fired when an item is being checked.
+ * @public
+ * @since 2.11.0
+ */
+@event("item-check", {
```

Comment:
@vladitasev @ilhan007,

Could you please take a look at this newly added event? I’m not sure about the naming and whether it’s being fired from the appropriate level. Should it be triggered at the group, menu, or item level?

Reply:
Just to mention that this event is used to signal the item's parent group that the item is checked/unchecked, so if the group is with single check mode, other checked items to be unchecked. Firing on group's level is useless in that case. Also, the check/uncheck (in fact click on item) is detected on Menu level where the target item is known, but as it does not know if it belongs to group or not (Menu also doesn't have this information), the event is fired by the target item and then bubbles up to parent group (if it exists) and the group does what's necessary.

Reply:
In addition, this event can be private as it is fired at the same time with `item-click` event. As I mentioned, it is used for internal signaling.